### PR TITLE
[RedTree] add `UserFeedbackCheckFailure` for `app_commands`

### PR DESCRIPTION
**Pull request by japandotorg**
### Description of the changes

Add a `UserFeedbackCheclkFailure` exception class with configurable message which is handled within `RedTree.on_error`

### Have the changes in this PR been tested?

No

### CODE REVIEW

File: redbot/core/tree.py [hunk 1]
```diff
@@ -331,6 +332,9 @@ async def on_error(
             await self._send_from_interaction(
                 interaction, _("You are not permitted to use this command.")
             )
+        elif isinstance(error, UserFeedbackCheckFailure):
+            if error.message:
+                await self._send_from_interaction(interaction, error.message)
```
Comment (Flame442) [inline redbot/core/tree.py]: This must be before the `CheckFailure` branch, or that branch will trigger first due to `UserFeedbackCheckFailure` being a subclass.

File: redbot/core/app_commands/errors.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"""Errors module for the app_commands package."""
+from discord import app_commands
+
+__all__ = ("UserFeedbackCheckFailure",)
```
Comment (Flame442) [inline redbot/core/app_commands/errors.py]: This package will not be public, so `__all__` should not be used. This class name should remain in the `__all__` of `app_commands.__init__`, as it is only guaranteed to exist when imported directly from `redbot.core.app_commands`.

> Lack of `__all__` in the module means that all of its names are considered private APIs.

https://docs.discord.red/en/stable/version_guarantees.html#developer-guarantees

**Review by Flame442 — CHANGES_REQUESTED** [review]

**Comment by japandotorg** [comment]
that should be it, requested changes have been made.

File: redbot/core/app_commands/errors.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+"""Errors module for the app_commands package."""
+from discord import app_commands
+
+
+class UserFeedbackCheckFailure(app_commands.CheckFailure):
+    """A generic version of CheckFailure."""
```
Comment (Jackenmen) [inline redbot/core/app_commands/errors.py]: The docstring from `commands.CheckFailure` makes more sense than this one:
```suggestion
    """A version of CheckFailure which isn't suppressed."""
```

Comment (japandotorg) [reply redbot/core/app_commands/errors.py]: i was gonna use that as well, but unlike `Red.on_command_error` `app_commands.CheckFailure` isn't really suppressed on `RedTree.on_error` https://github.com/Cog-Creators/Red-DiscordBot/blob/efc99c5dceda82537f75c24f734d4fb4dd4a8e11/redbot/core/tree.py#L336

Comment (japandotorg) [reply redbot/core/app_commands/errors.py]: would there be something more appropriate in this case?

Comment (Jackenmen) [reply redbot/core/app_commands/errors.py]: I see... Maybe something like this?
```suggestion
    """A version of CheckFailure responding with a custom error message."""
```

Comment (japandotorg) [reply redbot/core/app_commands/errors.py]: I suppose this would be a better one for now.

**Review by Flame442 — APPROVED** [review]
image

**Pull request merged**